src/lib.rs

```diff
@@ -228,10 +255,7 @@ pub struct LsColors {
     /// Whether Indicator::RegularFile falls back to Indicator::Normal
     /// (see <https://github.com/sharkdp/lscolors/issues/48#issuecomment-1582830387>)
     file_normal_fallback: bool,
-
-    // Note: you might expect to see a `HashMap` for `suffix_mapping` as well, but we need to
-    // preserve the exact order of the mapping in order to be consistent with `ls`.
```

Comment:
What happened to this?  The new implementation uses only a `HashMap` so I don't see how it can distinguish between `*.gz=01;33:*.tar.gz=01;31:` and `*.tar.gz=01;31:*.gz=01;33:`, which should behave differently.  (Perhaps we are missing a test if this wasn't caught.)

Reply:
I thought it was looking for the longest match, that's my bad, I'll fix it

Reply:
I think I fixed this, fixed the test also